orttraining/orttraining/core/graph/training_op_defs.cc

```diff
@@ -2473,6 +2474,67 @@ Return true if all elements are true and false otherwise.
           j++;
         }
       });
+
+  ONNX_CONTRIB_OPERATOR_SCHEMA(ATenOp)
+      .SetDomain(kMSDomain)
+      .SinceVersion(1)
+      .SetSupportLevel(OpSchema::SupportType::EXPERIMENTAL)
+      .SetDoc("ATenOp")
+      .Input(0, "inputs", "ATenOp inputs.", "T", OpSchema::Variadic,
+             /*is_homogeneous*/ false,
+             /*min_arity*/ 1)
+      .Output(0, "outputs", "ATenOp outputs.", "T", OpSchema::Variadic,
+              /*is_homogeneous*/ false,
+              /*min_arity*/ 1)
+      .Attr("name", "Name of ATen operator.", AttributeProto::STRING)
+      .Attr("custom_attributes_json", "custom attributes in JSON format.", AttributeProto::STRING, false)
+      .TypeConstraint("T", OpSchema::all_tensor_types(), "Allow inputs and outputs to be any kind of tensor.")
+      .TypeAndShapeInferenceFunction([](ONNX_NAMESPACE::InferenceContext& ctx) {
+        const auto name_proto = ctx.getAttribute("name");
+        ORT_ENFORCE(name_proto, "ATenOp's must have \"name\" attribute.");
+        const std::string& name = name_proto->s();
+        ORT_ENFORCE(contrib::aten_ops::ATEN_OPERATORS.find(name) != contrib::aten_ops::ATEN_OPERATORS.end());
+        contrib::aten_ops::ATenOperatorConfig op_config = contrib::aten_ops::ATEN_OPERATORS.at(name);
+        ORT_ENFORCE(ctx.getNumOutputs() == op_config.forward_tensor_output_type_configs.size());
+
+        // Set inferred output types.
+        for (size_t i = 0; i < ctx.getNumOutputs(); ++i) {
```

Comment:
I see... shape inference will start to fail starting from this point. 
One drawback of using ATenOps is that we will miss the shape inference on the forward graph, and it will result in ReduceSumTraining D2D memcpy.

Reply:
We are using symbolic shape for ORTModule so we cannot support the shape inference here. I added some logic to symbolic_shape_infer.py to do this.

Reply:
Could you help to have someone work on this as I am not quite familiar with exporter?